perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -884,6 +890,46 @@ private String getSQLWhereForMembersPage(MembersPageQuery query, MapSqlParameter
 		return " AND " + Utils.prepareSqlWhereForUserMemberSearch(query.getSearchString(), namedParams, false);
 	}
 
+	private String getSQLBasedOnPolicy() {
+		return "LEFT OUTER JOIN (SELECT gm.member_id, authz.role_id, authz.vo_id" +
+			"		FROM groups as g" +
+			"			JOIN authz ON g.id = authz.group_id" +
+			"			JOIN groups_members gm ON g.id = gm.group_id" +
+			"		 WHERE authz.user_id = (:userId))" +
+			"		AS members_group ON members.id = members_group.member_id AND members.vo_id = members_group.vo_id";
+	}
+
+	private String getWhereConditionBasedOnThePolicy(MembersPageQuery query, MapSqlParameterSource namedParams) {
+		PerunPolicy policy = AuthzResolverBlImpl.getPolicyByName("filter-getMembersPage_policy");
+		if (policy == null) {
+			return " WHERE members.vo_id = (:voId)";
+		}
+
+		StringBuilder sql = new StringBuilder();
+		sql.append(" WHERE members.vo_id = (:voId) AND (");
```

Comment:
Perhaps in the logical organization of the code having these two lines would be better later in the method as they might not be used if roles are evaluated empty.